alternator/executor.cc

```diff
@@ -148,6 +148,38 @@ std::string json_string::to_json() const {
     return _value;
 }
 
+// This function assumes the given value is an object and returns requested member value.
+// If it is not possible an api_error::validation is thrown.
+static const rjson::value& get_member(const rjson::value& obj, const char* member, const char* caller) {
+    if (!obj.IsObject()) {
+       throw api_error::validation(fmt::format("{} must be an object", caller));
+    }
+    const rjson::value* ret = rjson::find(obj, member);
+    if (!ret) {
+       throw api_error::validation(fmt::format("{} is missing a mandatory member {}", caller, member));
+    }
+    return *ret;
+}
+
+// This function assumes the given value is an object and returns requested member value that must be a non-empty object.
+// In case the requirements are not met an api_error::validation is thrown.
```

Comment:
Perhaps the word nonempty should appear in this function's name? It's not obvious that anyone that needs a member object also needs it to be non-empty.

UPDATE: Below I note that I think most of the callers shouldn't have the "non-empty" check at all. So maybe you should just drop the empyness check in this function? If one caller needs to check emptyness it can check it explicitly.